sqlmesh/core/engine_adapter/base.py

```diff
@@ -425,61 +517,98 @@ def _create_table_from_source_queries(
         columns_to_types: t.Optional[t.Dict[str, exp.DataType]] = None,
         exists: bool = True,
         replace: bool = False,
+        table_description: t.Optional[str] = None,
+        column_descriptions: t.Optional[t.Dict[str, str]] = None,
         **kwargs: t.Any,
     ) -> None:
         table = exp.to_table(table_name)
+
+        # Build a schema expression with column comments if the engine supports it and we have columns_to_types
+        schema = None
+        if (
+            column_descriptions
+            and self.COMMENT_CREATION.is_in_schema_def
+            and self.SUPPORTS_CTAS_SCHEMA_COMMENTS
+            and columns_to_types
+        ):
+            schema = self._build_schema_exp(table, columns_to_types, column_descriptions)
+
         with self.transaction(condition=len(source_queries) > 1):
             for i, source_query in enumerate(source_queries):
                 with source_query as query:
                     if i == 0:
                         self._create_table(
-                            table,
+                            schema if schema else table,
                             query,
                             columns_to_types=columns_to_types,
                             exists=exists,
                             replace=replace,
+                            table_description=table_description,
                             **kwargs,
                         )
                     else:
                         self._insert_append_query(
                             table_name, query, columns_to_types or self.columns(table)
                         )
 
+        # Register comments with commands if the engine doesn't support comments in the schema
+        if (
+            (table_description or column_descriptions)
+            and self.COMMENT_CREATION.is_comment_command_only
+        ) or (
+            column_descriptions
+            and self.COMMENT_CREATION.is_in_schema_def
+            and (not self.SUPPORTS_CTAS_SCHEMA_COMMENTS or not columns_to_types)
```

Comment:
I'm really confused by this one. If `SUPPORTS_CTAS_SCHEMA_COMMENTS` is True but `columns_to_types` are missing why do we still need a separate statement to create comments?

Overall this condition is really hairy and I suggest finding a way to rewrite it.

Reply:
To register the comments in the CTAS call for engines with `SUPPORTS_CTAS_SCHEMA_COMMENTS is True`, we have to build a full schema expression like `(a INTEGER COMMENT 'comment')`. 

`columns_to_types` is required to have the types for that schema expression, so if we don't have it available we need to use separate statements instead.

The `schema` object will be `None` unless we have already registered the comments in the CTAS call, so how about this?

```
        if (
            (table_description or column_descriptions)
            and self.COMMENT_CREATION.is_comment_command_only
        ) or (
            column_descriptions
            and not self.COMMENT_CREATION.is_unsupported
            and schema is None
        ):
```

Reply:
The simplest approach here would be to just always use the separate commands, but that results in one command per column comment.

Empirically, if columns_to_types will rarely be available then the benefits of comments in the schema won't ever accrue and this complexity is for nought. Do you have a sense for the proportion of CTAS calls that will have columns_to_types available?